docs/design/perfect_negotiation.md

```diff
@@ -0,0 +1,31 @@
+# Perfect Negotiation
+
+This document describes how perfect negotiation is implemented in OWT P2P SDK to avoid collision.
+
+**Perfect negotiation**, **polite peer**, **impolite peer** are defined in [Perfect negotiation example of WebRTC 1.0](https://w3c.github.io/webrtc-pc/#perfect-negotiation-example).
+
+## Determining polite peer and impolite peer
+
+OWT client SDKs determines polite peer and impolite peer by comparing client IDs. Sorting these two client IDs alphabetically in increasing order, the first one is polite order, the second one is impolite peer.
+
+Signaling server is required to return a client ID assigned for the client connected after authentication. OWT doesn't define how authentication works and how client IDs are assigned. They depend on application's design. But every client gets a unique client ID after connecting to the signaling server.
+
+## Connections
+
+We expect only one `PeerConnection` between two endpoints. A random ID is generated when creating a new `PeerConnection`. All messages (offers, answers, ICE candidates) for this `PeerConnection` have a `connectionId` property with the connection ID as its value.
```

Comment:
Please clarify if Client A and Client B sets up the peer connection, will connectionId value to be the same for both?

Reply:
Fixed. Connection ID is shared by both clients.